### PR TITLE
ci: enable check for static non-POD

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

*Description*:
Check global static non-POD in clang-tidy to avoid static initialization fiasco.
https://clang.llvm.org/extra/clang-tidy/checks/fuchsia-statically-constructed-objects.html

*Risk Level*: Low
*Testing*:
*Docs Changes*:
*Release Notes*:

### CODE REVIEW

File: include/envoy/singleton/manager.h [hunk 1]
```diff
@@ -45,7 +45,7 @@ template <const char* name_param> class RegistrationImpl : public Registration {
  */
 #define SINGLETON_MANAGER_REGISTRATION(NAME)                                                       \
   static constexpr char NAME##_singleton_name[] = #NAME "_singleton";                              \
-  static Envoy::Registry::RegisterFactory<                                                         \
+  static Envoy::Registry::RegisterFactory< /* NOLINT(fuchsia-statically-constructed-objects) */    \
```
Comment (lizan) [inline include/envoy/singleton/manager.h]: @mattklein123 we can do sth like this for extensions if we use macro to register, WDYT? cc @htuch

Comment (mattklein123) [reply include/envoy/singleton/manager.h]: I guess? In general this strikes me as solving a problem that doesn't actually need solving, but I guess if we insist on not having any non-POD statics in the code base we have to do ithis?

**Comment by htuch** [comment]
@lizan where are we with this PR?

**Comment by lizan** [comment]
@htuch PTAL, should be good to go.

**Pull request merged**